packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -265,12 +273,51 @@ export function createRouteManifest(
 
 	if (fs.existsSync(pages)) {
 		walk(fileURLToPath(pages), [], []);
-	} else {
+	} else if (config?._ctx?.injectedRoutes?.length === 0) {
 		const pagesDirRootRelative = pages.href.slice(config.root.href.length);
 
 		warn(logging, 'astro', `Missing pages directory: ${pagesDirRootRelative}`);
 	}
 
+	config?._ctx?.injectedRoutes?.forEach(({pattern: name, entryPoint}) => {
+		const resolved = require.resolve(entryPoint, { paths: [cwd || fileURLToPath(config.root)] });
```

Comment:
Is there a reason why we need to resolve this here? With renderers we don't do this, it gets resolved through Vite. I would think that's what we'd want to do here as well, if possible.

Reply:
An `entrypoint` can be a bare module specifier, e.g. `foo/my-file.astro`, but we need that to become `node_modules/foo/my-file.astro`

Reply:
Yeah, but I wonder if there's a better place to do this, so that Vite resolves this and not us. This would make it so aliases and all of those resolve features will work. But I don't think it's important to block this feature on this, so going to approve for now.